pulsar-client-cpp/lib/MultiTopicsBrokerConsumerStatsImpl.cc

```diff
@@ -0,0 +1,163 @@
+    statsList_.resize(size);
+}
+
+bool MultiTopicsBrokerConsumerStatsImpl::isValid() const {
+    bool isValid = true;
+    for (int i = 0; i < statsList_.size(); i++) {
+        isValid &= statsList_[i].isValid();
+    }
+    return isValid;
+}
+
+std::ostream& operator<<(std::ostream& os, const MultiTopicsBrokerConsumerStatsImpl& obj) {
+    os << "\nMultiTopicsBrokerConsumerStatsImpl ["
+       << "validTill_ = " << obj.isValid() << ", msgRateOut_ = " << obj.getMsgRateOut()
+       << ", msgThroughputOut_ = " << obj.getMsgThroughputOut()
+       << ", msgRateRedeliver_ = " << obj.getMsgRateRedeliver()
+       << ", consumerName_ = " << obj.getConsumerName()
+       << ", availablePermits_ = " << obj.getAvailablePermits()
+       << ", unackedMessages_ = " << obj.getUnackedMessages()
+       << ", blockedConsumerOnUnackedMsgs_ = " << obj.isBlockedConsumerOnUnackedMsgs()
+       << ", address_ = " << obj.getAddress() << ", connectedSince_ = " << obj.getConnectedSince()
+       << ", type_ = " << obj.getType() << ", msgRateExpired_ = " << obj.getMsgRateExpired()
+       << ", msgBacklog_ = " << obj.getMsgBacklog() << "]";
+    return os;
+}
+
+double MultiTopicsBrokerConsumerStatsImpl::getMsgRateOut() const {
+    double sum = 0;
+    for (int i = 0; i < statsList_.size(); i++) {
+        sum += statsList_[i].getMsgRateOut();
+    }
+    return sum;
+}
+
+double MultiTopicsBrokerConsumerStatsImpl::getMsgThroughputOut() const {
+    double sum = 0;
+    for (int i = 0; i < statsList_.size(); i++) {
+        sum += statsList_[i].getMsgThroughputOut();
+    }
+    return sum;
+}
+
+double MultiTopicsBrokerConsumerStatsImpl::getMsgRateRedeliver() const {
+    double sum = 0;
+    for (int i = 0; i < statsList_.size(); i++) {
+        sum += statsList_[i].getMsgRateRedeliver();
+    }
+    return sum;
+}
+
+const std::string MultiTopicsBrokerConsumerStatsImpl::getConsumerName() const {
+    std::string str;
+    for (int i = 0; i < statsList_.size(); i++) {
+        str += statsList_[i].getConsumerName() + DELIMITER;
+    }
+    return str;
+}
+
+uint64_t MultiTopicsBrokerConsumerStatsImpl::getAvailablePermits() const {
+    uint64_t sum = 0;
+    for (int i = 0; i < statsList_.size(); i++) {
+        sum += statsList_[i].getAvailablePermits();
+    }
+    return sum;
+}
+
+uint64_t MultiTopicsBrokerConsumerStatsImpl::getUnackedMessages() const {
+    uint64_t sum = 0;
+    for (int i = 0; i < statsList_.size(); i++) {
+        sum += statsList_[i].getUnackedMessages();
+    }
+    return sum;
+}
+
+bool MultiTopicsBrokerConsumerStatsImpl::isBlockedConsumerOnUnackedMsgs() const {
+    if (statsList_.size() == 0) {
+        return false;
+    }
+
+    bool isValid = true;
+    for (int i = 0; i < statsList_.size(); i++) {
+        isValid &= statsList_[i].isValid();
+    }
+    return isValid;
+}
+
+const std::string MultiTopicsBrokerConsumerStatsImpl::getAddress() const {
```

Comment:
Remove `const`

Reply:
Thanks. @merlimat,  the const in `const std::string`, or `const {)`?  not quite understand the reason of remove const?